test/e2e/storage/testsuites/multivolume.go

```diff
@@ -503,3 +527,62 @@ func TestConcurrentAccessToSingleVolume(f *framework.Framework, cs clientset.Int
 		utils.CheckReadFromPath(f, pod, *pvc.Spec.VolumeMode, path, byteLen, seed)
 	}
 }
+
+// getCurrentTopologies() goes through all Nodes and returns unique driver topologies and count of Nodes per topology
+func getCurrentTopologiesNumber(cs clientset.Interface, keys []string) ([]topology, []int, error) {
+	nodes, err := e2enode.GetReadySchedulableNodes(cs)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	topos := []topology{}
+	topoCount := []int{}
+
+	// TODO: scale?
+	for _, n := range nodes.Items {
+		topo := map[string]string{}
+		for _, k := range keys {
+			v, ok := n.Labels[k]
+			if !ok {
+				return nil, nil, fmt.Errorf("node %v missing topology label %v", n.Name, k)
```

Comment:
Should we fail if a certain node doesn't have a topology key? Don't  we want to skip instead?

Reply:
This behaviour is copied from the topology e2e tests. We only look for the keys that are expected to be there by the driver. So if some of the required topology keys is missing I think it's correct to fail the test.

Reply:
I was not sure what to do if the driver provides topology keys but the node misses them. I can just skip if that's safe.